wallet/src/de/schildbach/wallet/ui/username/voting/RequestUserNameViewModel.kt

```
@@ -437,4 +530,40 @@ class RequestUserNameViewModel @Inject constructor(
             it.createdAt
         }
     }
+
+    fun isUsingInvite(): Boolean = createUsernameArgs?.invite != null
+
+    suspend fun getInviteAssetLockTransaction(): AssetLockTransaction? = withContext(Dispatchers.IO) {
+        if (isUsingInvite()) {
+            inviteAssetLockTx.value = try {
+                topUpRepository.getAssetLockTransaction(createUsernameArgs?.invite!!)
+            } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)